src/preconditions/GuildOnly.ts

```diff
@@ -1,11 +1,23 @@
-import type { Message } from 'discord.js';
+import type { CommandInteraction, ContextMenuInteraction, Message } from 'discord.js';
 import { Identifiers } from '../lib/errors/Identifiers';
-import { Precondition, PreconditionResult } from '../lib/structures/Precondition';
+import { AllFlowsPrecondition } from '../lib/structures/Precondition';
 
-export class CorePrecondition extends Precondition {
-	public run(message: Message): PreconditionResult {
-		return message.guild === null
-			? this.error({ identifier: Identifiers.PreconditionGuildOnly, message: 'You cannot run this command in DMs.' })
+export class CorePrecondition extends AllFlowsPrecondition {
+	public messageRun(message: Message) {
+		return message.guildId === null
+			? this.error({ identifier: Identifiers.PreconditionGuildOnly, message: 'You cannot run this message command in DMs.' })
+			: this.ok();
+	}
+
+	public chatInputRun(interaction: CommandInteraction) {
+		return interaction.guildId === null
```

Comment:
Is the if check even worth having considering Chat input commands and Context menu commands cannot be in DMs to begin with? Wouldn't it be better to just always return an error? Or will that cause an error if someone has:
```ts
@ApplyOptions<Command.Options>({
	preconditions: ['GuildOnly']
})
export class MyExtendedCustomClass extends Command {
	
}
```
and then extends that for their commands, thus always pulling in this precondition.

Reply:
Chat input commands and context menu commands can be used in dms, mind you, only in DMs with the application, but the can be used!

Reply:
Oh okay. My bad. I thought that wasn't possible.